system-test/read-rows.ts

```diff
@@ -12,78 +12,123 @@
-    }
-  }
-}
-
-function rowResponse(rowKey: {}) {
+/**
+ * The read-rows-retry-test.json file contains row key data that represents
+ * data that the mock server should send back. This function converts this test
+ * data into responses that the mock server actually sends which match the
+ * format readrows responses should have.
+ *
+ * @param {string} rowKey The row key from read-rows-retry-test.json.
+ */
+function rowResponseFromServer(rowKey: string) {
   return {
-    rowKey: Mutation.convertToBytes(rowKey),
+    rowKey: Buffer.from(rowKey).toString('base64'),
     familyName: {value: 'family'},
-    qualifier: {value: 'qualifier'},
-    valueSize: 0,
-    timestampMicros: 0,
-    labels: [],
+    qualifier: {value: Buffer.from('qualifier').toString('base64')},
     commitRow: true,
-    value: 'value',
+    value: Buffer.from(rowKey).toString('base64'),
   };
 }
 
+function isRowKeysWithFunction(array: unknown): array is RowKeysWithFunction {
+  return (array as RowKeysWithFunction).asciiSlice !== undefined;
+}
+
+function isRowKeysWithFunctionArray(
+  array: unknown[]
+): array is RowKeysWithFunction[] {
+  return array.every((element: unknown) => {
+    return isRowKeysWithFunction(element);
+  });
+}
+
+interface TestRowRange {
+  startKey?: 'startKeyClosed' | 'startKeyOpen';
+  endKey?: 'endKeyOpen' | 'endKeyClosed';
+  startKeyClosed?: Uint8Array | string | null;
+  startKeyOpen?: Uint8Array | string | null;
+  endKeyOpen?: Uint8Array | string | null;
+  endKeyClosed?: Uint8Array | string | null;
+}
+interface RowKeysWithFunction {
+  asciiSlice: () => string;
+}
+function getRequestOptions(request: {
+  rows?: {
+    rowRanges?: TestRowRange[] | null;
+    rowKeys?: Uint8Array[] | null;
+  } | null;
+  rowsLimit?: string | number | Long | null | undefined;
+}): CreateReadStreamRequest {
+  const requestOptions = {} as CreateReadStreamRequest;
+  if (request.rows && request.rows.rowRanges) {
+    requestOptions.rowRanges = request.rows.rowRanges.map(
+      (range: TestRowRange) => {
+        const convertedRowRange = {} as {[index: string]: string};
+        {
+          // startKey and endKey get filled in during the grpc request.
+          // They should be removed as the test data does not look
+          // for these properties in the request.
+          if (range.startKey) {
+            delete range.startKey;
+          }
+          if (range.endKey) {
+            delete range.endKey;
+          }
+        }
+        Object.entries(range).forEach(
+          ([key, value]) => (convertedRowRange[key] = value.asciiSlice())
+        );
+        return convertedRowRange;
+      }
+    );
+  }
+  if (
+    request.rows &&
+    request.rows.rowKeys &&
+    isRowKeysWithFunctionArray(request.rows.rowKeys)
+  ) {
+    requestOptions.rowKeys = request.rows.rowKeys.map(
+      (rowKeys: RowKeysWithFunction) => rowKeys.asciiSlice()
+    );
+  }
+  // The grpc protocol sets rowsLimit to '0' if rowsLimit is not provided in the
```

Comment:
I think this is done just before the request is sent out in the grpc library. The main point here is that when the mock server receives the `rowsLimit`, if it was not provided then at this point it will be set to `'0'` so an explicit check needs to be done for that particular case. This is necessary in order to not have to change `read-rows-retry-test.json`.